src/socket-proxy/socket-proxyd.c

```diff
@@ -97,14 +97,8 @@ static void context_free(Context *context) {
 
         assert(context);
 
-        while ((es = set_steal_first(context->listen)))
-                sd_event_source_unref(es);
-
-        while ((c = set_first(context->connections)))
-                connection_free(c);
-
-        set_free(context->listen);
-        set_free(context->connections);
+        set_free_with_helper(context->listen, sd_event_source_unref);
+        set_free_with_helper(context->connections, connection_free);
```

Comment:
es and c is not used anymore.

Reply:
Ack. I don't know how I missed that, the compiler emits a warning of course.